src/main/java/org/jboss/jandex/Indexer.java

```diff
@@ -747,6 +757,30 @@ private void resolveTypeAnnotation(AnnotationTarget target, TypeAnnotationState
         }
     }
 
+    private boolean skipBridge(TypeAnnotationState typeAnnotationState, MethodInfo method) {
+        // javac copies annotations to bridge methods (which is good), however type annotations
+        // might become invalid. For instance, the bridge signature for @Nullable Object[] is
+        // Object (non-array), so usage=ARRAY signature is invalid. Other cases include annotated
+        // inner classes.
+        //
+        // We ignore those annotations for the bridge methods.
+        // See https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6695379
+        return isBridge(method) && typeAnnotationState.bridgeIncompatible;
```

Comment:
```suggestion
        return typeAnnotationState.bridgeIncompatible && isBridge(method);
```

The reorder of the conditions might slightly improve the performance  since `bridgeIncompatible` is slightly easier to verify